sdk/core/AzureCore/Source/Pipeline/Policies/HeadersValidationPolicy.swift

```diff
@@ -62,8 +62,9 @@ public class HeadersValidationPolicy: PipelineStage {
                     requestValue = "REDACTED"
                     responseValue = "REDACTED"
                 }
-                error = AzureError.general(
-                    "Value for header '\(key)' did not match. Expected: \(requestValue ?? "nil") Actual: \(responseValue)"
+                error = AzureError.sdk(
+                    "Value for header '\(key)' did not match. Expected: \(requestValue ?? "nil") Actual: \(responseValue)",
+                    nil
```

Comment:
Inner error is optional here, can omit the nil parameter.

Reply:
Missed that one when I was undoing adding `nil` to everything :)